packages/ai/src/lib/TldrawAiModule.ts

```diff
@@ -32,141 +25,87 @@ export class TldrawAiModule {
-				if (transform.transformChanges) {
-					changes = transform.transformChanges(changes)
-				}
-			}
+			apply({ change, editor: this.opts.editor })
 		}
 
 		return {
-			prompt: _prompt,
+			prompt,
 			handleChange,
-			handleChanges,
-		}
-	}
-
-	/**
-	 * Apply a change to the editor.
-	 *
-	 * @param change - The change to apply
-	 */
-	applyChange(change: TLAiChange) {
-		const { editor } = this.opts
-
-		if (editor.isDisposed) return
-
-		try {
-			switch (change.type) {
-				case 'createShape': {
-					editor.createShape(change.shape)
-					break
-				}
-				case 'updateShape': {
-					editor.updateShape(change.shape as TLShapePartial)
-					break
-				}
-				case 'deleteShape': {
-					editor.deleteShape(change.shapeId)
-					break
-				}
-				case 'createBinding': {
-					editor.createBinding(change.binding)
-					break
-				}
-				case 'updateBinding': {
-					editor.updateBinding(change.binding)
-					break
-				}
-				case 'deleteBinding': {
-					editor.deleteBinding(change.bindingId)
-					break
-				}
-				default:
-					exhaustiveSwitchError(change)
-			}
-		} catch (e) {
-			console.error('Error handling change:', e)
 		}
 	}
 
 	/**
-	 * Create the prompt to be sent to the AI.
+	 * Create a full prompt to be sent to the AI.
 	 *
-	 * @param prompt - The user's prompt
-	 * @param options - Options to generate the input
+	 * @param options - The options to generate the prompt
 	 */
-	async getPrompt(
-		prompt: TLAiMessages,
-		options = {} as Partial<Pick<TLAiPrompt, 'canvasContent' | 'contextBounds' | 'promptBounds'>>
-	): Promise<TLAiPrompt> {
+	async getPrompt(options: TldrawAiPromptOptions): Promise<TLAiPrompt> {
 		const { editor } = this.opts
-		const {
-			contextBounds = editor.getViewportPageBounds(),
-			promptBounds = editor.getViewportPageBounds(),
-		} = options
 
-		const content = options.canvasContent ?? this.getContent(promptBounds)
+		const _options = typeof options === 'string' ? { message: options } : options
 
-		// Get image from the content
-		const image = await this.getImage(content)
+		const contextBounds = _options.contextBounds ?? editor.getViewportPageBounds()
+		const promptBounds = _options.promptBounds ?? editor.getViewportPageBounds()
+		const canvasContent = _options.canvasContent ?? this.getContent(contextBounds)
+		const image = _options.image ?? (await this.getImage(canvasContent))
 
 		return {
-			message: asMessage(prompt),
-			canvasContent: content,
+			message: _options.message ?? '',
+			canvasContent,
 			contextBounds: roundBox(contextBounds),
 			promptBounds: roundBox(promptBounds),
 			image,
+			meta: _options.meta,
 		}
 	}
 
 	/**
 	 * Get the content from the current page.
 	 *
-	 * @param bounds - The bounds to get the content for
+	 * @param bounds - The bounds to get the content for. If not provided, includes the entire page.
 	 */
-	private getContent(bounds: Box): TLAiContent {
+	private getContent(bounds?: Box): TLAiContent {
 		const { editor } = this.opts
```

Comment:
If we don't need the full page option, let's remove this full-page thing. If we want to support full-page in the future, it's simple to add it back in.